### ISSUE TITLE
Meteor 2.6: Array field converted to an object in publication

### ISSUE THREAD

**Issue by ninjaPixel** [meteor/meteor#11887]
# Versions
Package versions: Meteor 2.6 and mongo 1.14.1

Operating system: MacOS (M1)

# Issue
Adding a new field to a document, with the following shape

```
foo: { bar: [1, 2, 3] }
```

results in the published `bar` array being converted to an object. i.e. it looks like this:

```
foo: { bar: {"0": 1, "1": 2, "2": 3} }
```

This issue occurs if the method is executed just server-side and also if the method is executed client-side.

Here is a simple app to reproduce the error: https://github.com/ninjaPixel/meteor-test-app

This issue also occurs when using mongo 1.14.0 (and that's as low as I'm allowed to drop the mongo package version while on Meteor 2.6).

# Last working version
This issue does not occur with Meteor 2.5.6 and mongo 1.13.0 (I'm unable to install mongo 1.14 within a Meteor 2.5.6 project).

Here is the same app, but with Meteor 2.5.6 and mongo 1.13.0 : https://github.com/ninjaPixel/meteor-test-app/tree/meteor-2-5-6-mongo-1-13

**Comment by renanccastro** [comment]
hi @ninjaPixel thanks for the detailed reproduction and issue description, that's really helpful.

Can you please give it a try at this branch: https://github.com/meteor/meteor/pull/11888 - [fix/array-hotfix-oplog](https://github.com/meteor/meteor/tree/fix/array-hotfix-oplog) and let me know the results, please?

**Comment by ninjaPixel** [comment]
Thanks for the patch, @renanccastro 
I can confirm that it fixes this issue. Nice one!

**Comment by renanccastro** [comment]
The fix for this issue is published on mongo@1.14.2!
